collectors/perf.go

```diff
@@ -55,240 +66,243 @@ func NewPerfCollector() *PerfCollector {
 			prometheus.BuildFQName(wmiNamespace, "perf", "avg_disk_bytes_per_transfer"),
 			"AvgDiskBytesPerTransfer",
-			nil,
+			[]string{"volume"},
 			nil,
 		),
 
 		AvgDiskBytesPerTransfer_Base: prometheus.NewDesc(
 			prometheus.BuildFQName(wmiNamespace, "perf", "avg_disk_bytes_per_transfer_base"),
 			"AvgDiskBytesPerTransfer_Base",
-			nil,
+			[]string{"volume"},
 			nil,
 		),
 
 		AvgDiskBytesPerWrite: prometheus.NewDesc(
 			prometheus.BuildFQName(wmiNamespace, "perf", "avg_disk_bytes_per_write"),
 			"AvgDiskBytesPerWrite",
-			nil,
+			[]string{"volume"},
 			nil,
 		),
 
 		AvgDiskBytesPerWrite_Base: prometheus.NewDesc(
 			prometheus.BuildFQName(wmiNamespace, "perf", "avg_disk_bytes_per_write_base"),
 			"AvgDiskBytesPerWrite_Base",
-			nil,
+			[]string{"volume"},
 			nil,
 		),
 
 		AvgDiskQueueLength: prometheus.NewDesc(
 			prometheus.BuildFQName(wmiNamespace, "perf", "avg_disk_queue_length"),
 			"AvgDiskQueueLength",
-			nil,
+			[]string{"volume"},
 			nil,
 		),
 
 		AvgDiskReadQueueLength: prometheus.NewDesc(
 			prometheus.BuildFQName(wmiNamespace, "perf", "avg_disk_read_queue_length"),
 			"AvgDiskReadQueueLength",
-			nil,
+			[]string{"volume"},
 			nil,
 		),
 
 		AvgDiskSecPerRead: prometheus.NewDesc(
 			prometheus.BuildFQName(wmiNamespace, "perf", "avg_disk_sec_per_read"),
 			"AvgDiskSecPerRead",
-			nil,
+			[]string{"volume"},
 			nil,
 		),
 
 		AvgDiskSecPerRead_Base: prometheus.NewDesc(
 			prometheus.BuildFQName(wmiNamespace, "perf", "avg_disk_sec_per_read_base"),
 			"AvgDiskSecPerRead_Base",
-			nil,
+			[]string{"volume"},
 			nil,
 		),
 
 		AvgDiskSecPerTransfer: prometheus.NewDesc(
 			prometheus.BuildFQName(wmiNamespace, "perf", "avg_disk_sec_per_transfer"),
 			"AvgDiskSecPerTransfer",
-			nil,
+			[]string{"volume"},
 			nil,
 		),
 
 		AvgDiskSecPerTransfer_Base: prometheus.NewDesc(
 			prometheus.BuildFQName(wmiNamespace, "perf", "avg_disk_sec_per_transfer_base"),
 			"AvgDiskSecPerTransfer_Base",
-			nil,
+			[]string{"volume"},
 			nil,
 		),
 
 		AvgDiskSecPerWrite: prometheus.NewDesc(
 			prometheus.BuildFQName(wmiNamespace, "perf", "avg_disk_sec_per_write"),
 			"AvgDiskSecPerWrite",
-			nil,
+			[]string{"volume"},
 			nil,
 		),
 
 		AvgDiskSecPerWrite_Base: prometheus.NewDesc(
 			prometheus.BuildFQName(wmiNamespace, "perf", "avg_disk_sec_per_write_base"),
 			"AvgDiskSecPerWrite_Base",
-			nil,
+			[]string{"volume"},
 			nil,
 		),
 
 		AvgDiskWriteQueueLength: prometheus.NewDesc(
 			prometheus.BuildFQName(wmiNamespace, "perf", "avg_disk_write_queue_length"),
 			"AvgDiskWriteQueueLength",
-			nil,
+			[]string{"volume"},
 			nil,
 		),
 
 		CurrentDiskQueueLength: prometheus.NewDesc(
 			prometheus.BuildFQName(wmiNamespace, "perf", "current_disk_queue_length"),
 			"CurrentDiskQueueLength",
-			nil,
+			[]string{"volume"},
 			nil,
 		),
 
 		DiskBytesPerSec: prometheus.NewDesc(
 			prometheus.BuildFQName(wmiNamespace, "perf", "disk_bytes_per_sec"),
 			"DiskBytesPerSec",
```

Comment:
Not from what I can find, at least. Most of the stuff that can be extracted from WMI is pre-aggregated in some way (some classes _only_ export running averages and percentages). I'll dig around a bit, but it doesn't look too good.